docs/architecture/ADR-001-interest-stream.md

```diff
@@ -0,0 +1,112 @@
+
+From section 2.1 of the [Umee Whitepaper](https://umee.cc/umee-whitepaper/):
+
+>upon deposit of assets into the Asset Facilities, users will receive an amount of tokens called uTokens that map 1:1 with the asset deposited. uTokens are initially minted on Umee and can bridge over to Ethereum as ERC20 tokens. The balance of uTokens grows over time by the underlying interest rate applied to the deposits. uTokens will employ an interest stream mechanism which means that a balance of uTokens will constantly generate income even when split or transferred.
+
+We need to find suitable ways to implement the interest stream on uTokens (which seems to require automatic minting or self-minting of tokens from existing balances). The method we choose should function in at least the following scenarios:
+- uToken balances held on Umee chain
+- uToken balances transferred or split between wallets on Umee chain
+- uToken balances sent to other Cosmos chains via IBC and held
+- uToken balances split or transferred while on other Cosmos chains
+- uToken balances sent from Umee -> a Cosmos chain -> a different Cosmos chain, then held, split, or transferred
+- uToken balances held on Ethereum as ERC20
+- uToken balances split or transferred on Ethereum as ERC20
+
+This collection of scenarios combined with the underlying implementation of Cosmos IBC, will create implementation challenges for the interest stream feature.
+
+From [Cosmos IBS tutorial](https://tutorials.cosmos.network/understanding-ibc-denoms/):
+
+> The value that tokens represent can be transferred across chains, but the token itself cannot. When sending the tokens with IBC to another blockchain:
+>
+> - Blockchain A locks the tokens and relays proof to blockchain B
+> - Blockchain B mints its own representative tokens in the form of voucher replacement tokens
+> - Blockchain B sends the voucher tokens back to blockchain A
+> - The voucher tokens are destroyed (burned) on blockchain B
+> - The locked tokens on blockchain A are unlocked
+>
+> The only way to unlock the locked tokens on blockchain A is to send the voucher token back from blockchain B. The result is that the voucher token on blockchain B is burned. The burn process purposefully takes the tokens out of circulation.
+
+This means that if uTokens are to be sent to other Cosmos blockchains, then the interest stream must apply equally to 'voucher tokens' on other chains. These chains are not likely to be running our code, so it is unclear how we would cause uToken balances sent vie IBC to generate interest.
+
+As an alternative, transfer of uTokens via IBC could be forbidden or unsupported - in that case, only the scenarios where uToken balances are held on Umee and Ethereum need to be addressed.
+
+## Decision
+
+"uToken to base asset exchange rate grows over time". This method is inspired by the Compound model.
```

Comment:
Maybe add a link to the model?

```suggestion
"uToken to base asset exchange rate grows over time". This method is inspired by the Compound [model](<LINK>).
```

Reply:
Note: One candidate link that might be used is [this example](https://compound.finance/docs/ctokens#introduction)

Reply:
Yup - Use whatever you feel is good enough 👍